float32 minRange
float32 maxRange
float32[] ranges
